src/sidebar/directive/test/login-control-test.js

```diff
@@ -0,0 +1,168 @@
+      var el = createLoginControl({
+        auth: {
+          status: 'unknown',
+        },
+        newStyle: true,
+      });
+      
+      var loginText = el[0].querySelectorAll('.login-text');
+      var menuLinks = el[0].querySelectorAll('.login-control-menu .dropdown-menu a');
+
+      assert.include(loginText[0].textContent, '⋯');
+      assert.equal(menuLinks.length, 1);
+      assert.equal(menuLinks[0].textContent, 'Help');
+    });
+  });
+
+  context('when user is logged out', function () {
+    it('only shows the help menu option', function () {
+      var el = createLoginControl({
+        auth: {
+          status: 'logged-out',
+        },
+        newStyle: true,
+      });
+
+      var loginText = el[0].querySelectorAll('.login-text a');
+      var menuLinks = el[0].querySelectorAll('.login-control-menu .dropdown-menu a');
+
+      assert.equal(loginText[0].textContent, 'Sign up');
+      assert.equal(loginText[1].textContent, 'Log in');
+      assert.equal(menuLinks.length, 1);
+      assert.equal(menuLinks[0].textContent, 'Help');
+    });
+  });
+
+  context('when a H user is logged in', function () {
+    it('shows the complete list of menu options', function () {
+      var el = createLoginControl({
+        auth: {
+          username: 'someUsername',
+          status: 'logged-in',
+        },
+        newStyle: true,
+      });
+
+      var menuLinks = el[0].querySelectorAll('.login-control-menu .dropdown-menu a');
+
+      assert.equal(menuLinks.length, 4);
+      assert.equal(menuLinks[0].textContent, 'someUsername');
+      assert.equal(menuLinks[1].textContent, 'Account settings');
+      assert.equal(menuLinks[2].textContent, 'Help');
+      assert.equal(menuLinks[3].textContent, 'Log out');
+    });
+  });
+
+  context('when a third party user is logged in', function () {
+    it('shows the help menu option and the username', function () {
+      var el = createLoginControl({
+        auth: {
+          userid: 'acct:someUsername@anotherFakeDomain',
+          username: 'someUsername',
+          status: 'logged-in',
+        },
+        newStyle: true,
+      });
+
+      var loginControlMenu = el[0].querySelector('.login-control-menu');
+      var menuLinks = loginControlMenu.querySelector('.dropdown-menu').querySelectorAll('a');
+      var disabledMenuItems =
+        loginControlMenu.querySelectorAll('.dropdown-menu__link--disabled');
+
+      assert.equal(menuLinks.length, 1);
+      assert.equal(menuLinks[0].textContent, 'Help');
+      assert.equal(disabledMenuItems.length, 1);
+      assert.include(disabledMenuItems[0].textContent, 'someUsername');
+    });
+  });
+
+  context('old controls when a H user is logged in', function () {
+    it('shows the complete list of menu options', function () {
+      var el = createLoginControl({
+        auth: {
+          username: 'someUsername',
+          status: 'logged-in',
+        },
+        newStyle: false,
```

Comment:
Its nice that you've written tests for this, I wonder though if we're in a position where we could get rid of support for the 'old style' soon.